### PR TITLE
add tox and switch to GithubAction

### PR DESCRIPTION
**Pull request by vincentsarago**
👋 This pr does: 
- switch to GH actions 🤷‍♂️ 
- add TOX for testing multiple python versions 🤷‍♂️ 

To be discussed 
- pre-commit integration?
- pypi publish from CI? 
- mkdoc / gh-page?

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        id: aws_creds
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
```
Comment (vincentsarago) [inline .github/workflows/ci.yml:28]: I guess you need AWS credential to be able to use S3 file 🤷‍♂️

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      - name: Upload Results
+        if: success()
+        uses: codecov/codecov-action@v1
+        with:
+          file: ./coverage.xml
+          flags: unittests
+          name: ${{ matrix.platform }}-${{ matrix.tox-env }}
+          fail_ci_if_error: false
+
+  # publish:
```
Comment (vincentsarago) [inline .github/workflows/ci.yml:53]: TDB

File: tox.ini [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+envlist = py37,py38
+
+[testenv]
+extras = dev
+commands=
+    python -m pytest --cov aiocogeo --cov-report xml --cov-report term-missing
+deps=
+    numpy
+passenv=
+    AWS_*
```
Comment (vincentsarago) [inline tox.ini:11]: we can remove this if S3 credential is not needed

**Comment by vincentsarago** [comment]
when trying to run the test within tox I get 
```
ImportError while loading conftest '/Users/vincentsarago/Workspace/vincentsarago/aiocogeo/tests/conftest.py'.
tests/conftest.py:9: in <module>
    from aiocogeo import config
aiocogeo/__init__.py:1: in <module>
    from .cog import COGReader, CompositeReader
aiocogeo/cog.py:16: in <module>
    from .filesystems import Filesystem
aiocogeo/filesystems.py:19: in <module>
    import aioboto3
.tox/py37/lib/python3.7/site-packages/aioboto3/__init__.py:5: in <module>
    from aioboto3.session import Session
.tox/py37/lib/python3.7/site-packages/aioboto3/session.py:9: in <module>
    import aiobotocore.session
.tox/py37/lib/python3.7/site-packages/aiobotocore/__init__.py:1: in <module>
    from .session import get_session, AioSession
.tox/py37/lib/python3.7/site-packages/aiobotocore/session.py:11: in <module>
    from .credentials import create_credential_resolver, AioCredentials
.tox/py37/lib/python3.7/site-packages/aiobotocore/credentials.py:15: in <module>
    from botocore.credentials import EnvProvider, Credentials, RefreshableCredentials, \
E   ImportError: cannot import name 'SSOProvider' from 'botocore.credentials' (/Users/vincentsarago/Workspace/vincentsarago/aiocogeo/.tox/py37/lib/python3.7/site-packages/botocore/credentials.py)
ERROR: InvocationError for command /Users/vincentsarago/Workspace/vincentsarago/aiocogeo/.tox/py37/bin/python -m pytest --cov aiocogeo --cov-report xml --cov-report term-missing (exited with code 4)
```
😬

File: setup.py [hunk 4]
```diff
@@ -14,8 +14,6 @@
         "pytest-asyncio<0.11.0",
         "pytest-cov",
         "shapely",
-        "botocore==1.15.32",
-        "boto3==1.12.32",
```
Comment (vincentsarago) [inline setup.py:18]: aioboto3, depends on aiobotocore which depends on specific boto3 and botocore versions. It's easier to let hem decide the version they need ;-)

File: tox.ini [hunk 5]
```diff
@@ -7,8 +7,6 @@ commands=
     python -m pytest --cov aiocogeo --cov-report xml --cov-report term-missing
 deps=
     numpy
-passenv=
-    AWS_*
```
Comment (vincentsarago) [inline tox.ini]: I think this is not needed in fact!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (geospatial-jeff) [reply .github/workflows/ci.yml]: Yeah should really use moto so we don't need to talk to S3 at all for tests

**Comment by geospatial-jeff** [comment]
I pulled down your changes, ran `tox` and it worked 🎉

**Comment by kylebarron** [comment]
You might be able to merge this onto a non-`master` branch, and then CI might run?

**Comment by geospatial-jeff** [comment]
Finished in #99 !

**Pull request closed without merging**